phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/engines/EnginesFilter.java

```diff
@@ -0,0 +1,88 @@
+package org.phoenicis.javafx.views.mainwindow.engines;
+
+import org.phoenicis.engines.dto.EngineSubCategoryDTO;
+import org.phoenicis.engines.dto.EngineVersionDTO;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.function.Predicate;
+
+/**
+ * This class represents a filter used for filtering the engine versions for
+ * <ul>
+ * <li>
+ * Installed engine versions
+ * </li>
+ * <li>
+ * Uninstalled engine versions
+ * </li>
+ * <li>
+ * Engine versions containing a search term
+ * </li>
+ * </ul>
+ * This filter depends on a previously defines {@link EngineSubCategoryDTO}.
+ *
+ * @author marc
+ * @since 23.04.17
+ */
+public class EnginesFilter implements Predicate<EngineVersionDTO> {
+    private EngineSubCategoryDTO engineSubCategory;
+
+    private String wineEnginesPath;
+
+    /**
+     * The search term entered into the search field
+     */
+    private String searchTerm = "";
+
+    /**
+     * Are only installed engines searched
+     */
+    private boolean onlyInstalled = true;
```

Comment:
The name is a bit misleading: it suggests that you can filter either installed or uninstalled (because of the "only"). If I understand this correctly, you can also search both. Maybe something like `showInstalled` would be better.